src/coordination/coordinator_instance.cpp

```diff
@@ -115,6 +63,78 @@ CoordinatorInstance::CoordinatorInstance(CoordinatorInstanceInitConfig const &co
     auto &repl_instance = self->FindReplicationInstance(repl_instance_name);
     std::invoke(repl_instance.GetFailCallback(), self, repl_instance_name);
   };
+
+  raft_state_ = std::make_unique<RaftState>(config, GetBecomeLeaderCallback(), GetBecomeFollowerCallback());
+  raft_state_->InitRaftServer();
```

Comment:
Why not Init call from the constructor instead of explicit call?

Reply:
Because there is the problem of `InitServer` calling coordinator `BecomeLeader` callback while `raft_state_` unique_ptr is not constructed still due to how std::make_unique works. It first calls constructor and only then initializes pointer